sklearn/utils/validation.py

```diff
@@ -982,3 +982,121 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
 
     if max_val is not None and x > max_val:
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
+
+
+def check_kernel_eigenvalues(lambdas, warn_on_zeros=False):
+    """Checks that the provided array of kernel eigenvalues for numerical or
+    conditioning issues and returns a fixed validated version.
+
+    It checks:
+
+     - that there are no significant imaginary parts in eigenvalues (more than
+     1e-5 times the maximum real part). If this check fails, it raises a
+     ValueError. Otherwise all non-significant imaginary parts that may remain
+     are removed.
+
+     - that eigenvalues are not all negative. If this check fails, it raises a
+     ValueError
+
+     - that there are no significant negative eigenvalues (with absolute value
+     more than 1e-10 and more than 1e-5 times the largest positive eigenvalue).
+     If this check fails, it raises a KernelWarning. All negative eigenvalues
+     are set to zero in all cases.
+
+     - that the eigenvalues are well conditioned. That means, that the
+     eigenvalues are all greater than the maximum eigenvalue divided by 1e12.
+     If this check fails and `warn_on_zeros=True`, it raises a KernelWarning.
+     All the eigenvalues that are too small are then set to zero.
+
+    Note: the returned array is converted to numpy array.
+
+    >>> check_kernel_eigenvalues((1, 2))      # nominal case
```

Comment:
Please follow numpy: put this under an Examples heading below the Returns section

Reply:
ok